### ISSUE TITLE
[BUG] Mining to MiningRigRentals with no hashrate

### ISSUE THREAD

**Issue by RainbowMiner** [sp-hash/TeamBlackMiner#17]
Looks like there is a problem with MiningRigRentals. Miner starts, logs into pool, but then - nothing. Gminer and Trex both run fine on that rental. So it's not a renter's problem.

Commandline (I have x-ed out my credentials: xxx = my username, 1234567 = rental id)
```
TBMiner.exe --algo etchash --hostname eu-01.miningrigrentals.com --port 3344 --wallet xxx.1234567 --worker_name rbm --server_passwd x --cuda-devices [0,1,2,3,4,5] --xintensity -1 --no-ansi --no-cpu
```

Log:
```
     0m [2021-09-13 23:09:08.155] OpenCL driver version: 27.20
     0m [2021-09-13 23:09:08.155] Cuda driver version: 11.4
     0m [2021-09-13 23:09:08.155] Cuda runtime API version: 11.4
     0m [2021-09-13 23:09:08.155] 
     0m [2021-09-13 23:09:08.155]                                 
     0m [2021-09-13 23:09:08.155] 
     0m [2021-09-13 23:09:08.155] Welcome to Team Black Miner 1.04
     0m [2021-09-13 23:09:08.155]                                 
     0m [2021-09-13 23:09:08.155] 
     0m [2021-09-13 23:09:08.468] GPU0 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:08.718] GPU1 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:08.983] GPU2 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:09.125] GPU3 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:09.280] GPU4 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:09.452] GPU5 NVIDIA GeForce GTX 1070 8GB (compute 6.1) added to miner as Cuda device
     0m [2021-09-13 23:09:09.515] Trying eu-01.miningrigrentals.com:3344 (etchash)
     0m [2021-09-13 23:09:09.639] Pool responded successfully to extranonce subscribe
     0m [2021-09-13 23:09:09.671] GPU0 Started Cuda work thread
     0m [2021-09-13 23:09:09.983] Pool responded successfully to authorize
     0m [2021-09-13 23:09:10.030] GPU1 Started Cuda work thread
     0m [2021-09-13 23:09:10.219] GPU2 Started Cuda work thread
     0m [2021-09-13 23:09:10.436] GPU3 Started Cuda work thread
     0m [2021-09-13 23:09:10.670] GPU4 Started Cuda work thread
     0m [2021-09-13 23:09:10.967] GPU5 Started Cuda work thread
     0m [2021-09-13 23:09:11.312] GPU3 Starting generating etchash dag (Epoch: 2147483647)
     0m [2021-09-13 23:09:11.312] GPU2 Starting generating etchash dag (Epoch: 2147483647)
     0m [2021-09-13 23:09:11.312] GPU0 Starting generating etchash dag (Epoch: 2147483647)
     0m [2021-09-13 23:09:11.312] GPU5 Starting generating etchash dag (Epoch: 2147483647)
     0m [2021-09-13 23:09:11.327] GPU4 Starting generating etchash dag (Epoch: 2147483647)
     0m [2021-09-13 23:09:11.436] GPU1 Starting generating etchash dag (Epoch: 2147483647)
     1m [2021-09-13 23:10:11.531] 
     1m [2021-09-13 23:10:11.531] 
     1m [2021-09-13 23:10:11.531] com (etchash)    PING: 0ms    DIFFICULTY: 0.000    EPOCH: 2147483647
     1m [2021-09-13 23:10:11.531]                                                                     
     1m [2021-09-13 23:10:11.531] 
     1m [2021-09-13 23:10:11.531] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     1m [2021-09-13 23:10:11.531] GPU0	1070	64/0	50	1982	4404	134	0.13	inf
     1m [2021-09-13 23:10:11.531] GPU1	1070	68/0	66	1942	4404	134	0.13	inf
     1m [2021-09-13 23:10:11.531] GPU2	1070	68/0	66	1937	4404	113	0.11	inf
     1m [2021-09-13 23:10:11.531] GPU3	1070	68/0	70	1893	4404	131	0.13	inf
     1m [2021-09-13 23:10:11.531] GPU4	1070	64/0	50	1923	4404	135	0.14	inf
     1m [2021-09-13 23:10:11.531] GPU5	1070	66/0	58	1951	4404	134	0.13	inf
     1m [2021-09-13 23:10:11.531] 						781	0.78	inf
     1m [2021-09-13 23:10:11.531] 
     1m [2021-09-13 23:10:11.531] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     1m [2021-09-13 23:10:11.531] GPU0	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] GPU1	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] GPU2	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] GPU3	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] GPU4	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] GPU5	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531] 			0.00 H/s	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-13 23:10:11.531]                                                                     
     1m [2021-09-13 23:10:11.531] 
     2m [2021-09-13 23:11:13.389] 
     2m [2021-09-13 23:11:13.389] 
     2m [2021-09-13 23:11:13.389] com (etchash)    PING: 0ms    DIFFICULTY: 0.000    EPOCH: 2147483647
     2m [2021-09-13 23:11:13.389]                                                                     
     2m [2021-09-13 23:11:13.389] 
     2m [2021-09-13 23:11:13.389] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     2m [2021-09-13 23:11:13.389] GPU0	1070	64/0	50	1967	4404	134	0.13	inf
     2m [2021-09-13 23:11:13.389] GPU1	1070	68/0	66	1959	4404	123	0.12	inf
     2m [2021-09-13 23:11:13.389] GPU2	1070	68/0	70	1907	4404	135	0.14	inf
     2m [2021-09-13 23:11:13.389] GPU3	1070	69/0	71	1871	4404	133	0.13	inf
     2m [2021-09-13 23:11:13.389] GPU4	1070	64/0	50	1941	4404	132	0.13	inf
     2m [2021-09-13 23:11:13.389] GPU5	1070	66/0	58	1939	4404	134	0.13	inf
     2m [2021-09-13 23:11:13.389] 						791	0.79	inf
     2m [2021-09-13 23:11:13.389] 
     2m [2021-09-13 23:11:13.389] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     2m [2021-09-13 23:11:13.389] GPU0	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] GPU1	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] GPU2	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] GPU3	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] GPU4	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] GPU5	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389] 			0.00 H/s	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     2m [2021-09-13 23:11:13.389]                                                                     
     2m [2021-09-13 23:11:13.389] 
     3m [2021-09-13 23:12:15.642] 
     3m [2021-09-13 23:12:15.642] 
     3m [2021-09-13 23:12:15.642] com (etchash)    PING: 0ms    DIFFICULTY: 0.000    EPOCH: 2147483647
     3m [2021-09-13 23:12:15.642]                                                                     
     3m [2021-09-13 23:12:15.642] 
     3m [2021-09-13 23:12:15.642] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     3m [2021-09-13 23:12:15.642] GPU0	1070	64/0	50	1968	4404	65	0.07	inf
     3m [2021-09-13 23:12:15.642] GPU1	1070	68/0	66	1943	4404	127	0.13	inf
     3m [2021-09-13 23:12:15.642] GPU2	1070	68/0	70	1909	4404	134	0.13	inf
     3m [2021-09-13 23:12:15.642] GPU3	1070	69/0	70	1876	4404	132	0.13	inf
     3m [2021-09-13 23:12:15.642] GPU4	1070	65/0	50	1963	4404	135	0.14	inf
     3m [2021-09-13 23:12:15.642] GPU5	1070	66/0	58	1955	4404	134	0.13	inf
     3m [2021-09-13 23:12:15.642] 						727	0.73	inf
     3m [2021-09-13 23:12:15.642] 
     3m [2021-09-13 23:12:15.642] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     3m [2021-09-13 23:12:15.642] GPU0	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] GPU1	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] GPU2	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] GPU3	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] GPU4	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] GPU5	1070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642] 			0.00 H/s	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     3m [2021-09-13 23:12:15.642]                                                                     
     3m [2021-09-13 23:12:15.642] 
     4m [2021-09-13 23:13:17.190] 
     4m [2021-09-13 23:13:17.190] 
     4m [2021-09-13 23:13:17.190] com (etchash)    PING: 0ms    DIFFICULTY: 0.000    EPOCH: 2147483647
     4m [2021-09-13 23:13:17.190]                                                                     
     4m [2021-09-13 23:13:17.190] 
     4m [2021-09-13 23:13:17.190] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     4m [2021-09-13 23:13:17.190] GPU0	1070	64/0	50	1986	4404	123	0.12	inf
```

**Comment by sp-hash** [comment]
miningrigrentals.com is assumed to use ethereum stratum v1.0. but perhaps its Ethproxy protocol?

**Comment by RainbowMiner** [comment]
This is what I have been using in the past: qtminer protocol
- Phoenix miner : `-proto 3`
- Trex: `stratum1+tcp://`

Maybe that helps.

**Comment by sp-hash** [comment]
The protocol is fine,  the problem is that the miner is unable to calculate the correct epoc
 (Epoch: 2147483647) in the output. We will fix it

**Comment by binaryf** [comment]
--wallet xxx.1234567 --worker_name rbm

We recommend using --worker_name.
Worker name will be added to username here.

So the "wallet-string" here will be xxx.1234567.rbm which could be right for miningrigrentals.

I will take another look. I am not sure how this works.
Do you get work from the pool when somebody rent your rig ?

**Comment by RainbowMiner** [comment]
> Do you get work from the pool when somebody rent your rig ?

Didn't look like. I cannot try, currently, because all my rigs are rented out with some Equihash algorithms.

**Comment by RainbowMiner** [comment]
The problem till persists with v1.05:

 ```
    0m [2021-09-19 09:14:56.806] OpenCL driver version: 27.20
     0m [2021-09-19 09:14:56.806] Cuda driver version: 11.4
     0m [2021-09-19 09:14:56.806] Cuda runtime API version: 11.4
     0m [2021-09-19 09:14:56.806] 
     0m [2021-09-19 09:14:56.806]                                 
     0m [2021-09-19 09:14:56.806] 
     0m [2021-09-19 09:14:56.806] Welcome to Team Black Miner 1.05
     0m [2021-09-19 09:14:56.806]                                 
     0m [2021-09-19 09:14:56.806] 
     0m [2021-09-19 09:14:57.087] GPU6 NVIDIA GeForce RTX 3070 8GB (compute 8.6) added to miner as Cuda device
     0m [2021-09-19 09:14:57.196] Trying eu-01.miningrigrentals.com:3344 (etchash)
     0m [2021-09-19 09:14:57.308] Pool responded successfully to extranonce subscribe
     0m [2021-09-19 09:14:57.680] Pool responded successfully to authorize
     0m [2021-09-19 09:14:57.696] GPU6 Started Cuda work thread
     0m [2021-09-19 09:14:58.680] GPU6 Starting generating etchash dag (Epoch: 2147483647)
     1m [2021-09-19 09:15:59.732] 
     1m [2021-09-19 09:15:59.732] 
     1m [2021-09-19 09:15:59.732] com (etchash)    PING: 0ms    DIFFICULTY: 0.000    EPOCH: 2147483647
     1m [2021-09-19 09:15:59.732]                                                                     
     1m [2021-09-19 09:15:59.732] 
     1m [2021-09-19 09:15:59.732] ID	BOARD	TEMP	FAN	CORE	MEM	WATT	kW/h	COST/h
     1m [2021-09-19 09:15:59.732] GPU6	3070	63/0	50	1146	7993	131	0.13	0.01$
     1m [2021-09-19 09:15:59.732] 						131	0.13	0.01$
     1m [2021-09-19 09:15:59.732] 
     1m [2021-09-19 09:15:59.732] ID	BOARD	TYPE	HASHRATE/W	HASHRATE	AVERAGE		SHARES
     1m [2021-09-19 09:15:59.732] GPU6	3070	Cuda	0.00 H/W	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-19 09:15:59.732] 			0.00 H/s	0.00 H/s	0.00 H/s	0/0/0 (100.00)
     1m [2021-09-19 09:15:59.732]                                                                     
     1m [2021-09-19 09:15:59.732] 
```

No jobs, strange Epoch, no hashrate.

**Comment by RainbowMiner** [comment]
Actually, 2147483647 is the maximum signed integer (64bit). Does this help to find the problem?

**Comment by sp-hash** [comment]
yes. it doesn't get work from the pool. We calculate the epoc out of the job coming from the pool.

**Comment by RainbowMiner** [comment]
> yes. it doesn't get work from the pool. We calculate the epoc out of the job coming from the pool. So I guess nobody has rented the rig yet?

No, the rig has been rented, when I tried. Is this fixed in v1.07?

**Comment by binaryf** [comment]
I will have another look at this tomorrow.
If you can explain how it is supposed to work, that would be nice.
I am very new to this with renting out capacity.
